python-ecosys/urequests/urequests.py

```diff
@@ -55,7 +71,18 @@ def request(method, url, data=None, json=None, headers={}, stream=None):
     ai = usocket.getaddrinfo(host, port, 0, usocket.SOCK_STREAM)
     ai = ai[0]
 
-    s = usocket.socket(ai[0], ai[1], ai[2])
+    resp_d = None
+    if parse_headers is not False:
+        resp_d = {}
+
+    s = usocket.socket(ai[0], usocket.SOCK_STREAM, ai[2])
+
+    if timeout is not None:
+        try:
+            s.settimeout(timeout)
+        except AttributeError:
+            raise AttributeError("Socket does not support timeout on this platform")
```

Comment:
I don't think it makes sense to convert the error string.  It'll already raise a sensible message.